File: app/src/main/java/com/microlabs/trallet/model/Category.java
package com.microlabs.trallet.model;

import io.realm.Realm;
import io.realm.RealmObject;

/**
 * Category Class
 */
public class Category extends RealmObject {

    private int id;
    private String name;

    public static final String fId = "id";
    public static final String fName = "name";

    public int getId() {
        return id;
    }

    public void setId(Realm realm) {
        id = realm.where(Category.class).max(Category.fId).intValue() + 1;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: app/src/main/java/com/microlabs/trallet/model/Category.kt
package com.microlabs.trallet.model

import androidx.annotation.DrawableRes
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "categories")
data class Category(
        @PrimaryKey(autoGenerate = true)
        val id: Int = 0,
        val name: String,
        @ColumnInfo(name = "imageId")
        @DrawableRes
        val imgId: Int
)